tools/clang/test/HLSLFileCheck/hlsl/workgraph/common_input_parameters.hlsl

```diff
@@ -0,0 +1,63 @@
+// CHECK:  %[[annotHdl:.+]] = call %dx.types.NodeRecordHandle @dx.op.annotateNodeRecordHandle(i32 251, %dx.types.NodeRecordHandle %[[Hdl]], %dx.types.NodeRecordInfo { i32 69, i32 40 })  ; AnnotateNodeRecordHandle(noderecord,props)
+// CHECK:  %[[node_ptr:.+]] = call %struct.RECORD.0 addrspace(6)* @dx.op.getNodeRecordPtr.struct.RECORD.0(i32 239, %dx.types.NodeRecordHandle %[[annotHdl]], i32 0)  ; GetNodeRecordPtr(recordhandle,arrayIndex)
+// CHECK:  %[[ptr:.+]] = getelementptr inbounds %struct.RECORD.0, %struct.RECORD.0 addrspace(6)* %[[node_ptr]], i32 0, i32 0, i32 0
+// CHECK:  store i32 %[[tid_x]], i32 addrspace(6)* %[[ptr]], align 4
+// CHECK:  %[[ptr:.+]] = getelementptr inbounds %struct.RECORD.0, %struct.RECORD.0 addrspace(6)* %[[node_ptr]], i32 0, i32 0, i32 1
+// CHECK:  store i32 %[[tid_y]], i32 addrspace(6)* %[[ptr]], align 4
+// CHECK:  %[[ptr:.+]] = getelementptr inbounds %struct.RECORD.0, %struct.RECORD.0 addrspace(6)* %[[node_ptr]], i32 0, i32 0, i32 2
+// CHECK:  store i32 %[[tid_z]], i32 addrspace(6)* %[[ptr]], align 4
+// CHECK:  %[[ptr:.+]] = getelementptr inbounds %struct.RECORD.0, %struct.RECORD.0 addrspace(6)* %[[node_ptr]], i32 0, i32 1, i32 0
+// CHECK:  store i32 %[[gid_x]], i32 addrspace(6)* %[[ptr]], align 4
+// CHECK:  %[[ptr:.+]] = getelementptr inbounds %struct.RECORD.0, %struct.RECORD.0 addrspace(6)* %[[node_ptr]], i32 0, i32 1, i32 1
+// CHECK:  store i32 %[[gid_y]], i32 addrspace(6)* %[[ptr]], align 4
+// CHECK:  %[[ptr:.+]] = getelementptr inbounds %struct.RECORD.0, %struct.RECORD.0 addrspace(6)* %[[node_ptr]], i32 0, i32 1, i32 2
+// CHECK:  store i32 %[[gid_z]], i32 addrspace(6)* %[[ptr]], align 4
+// CHECK:  %[[ptr:.+]] = getelementptr %struct.RECORD.0, %struct.RECORD.0 addrspace(6)* %[[node_ptr]], i32 0, i32 2
+// CHECK:  store i32 %[[ftid]], i32 addrspace(6)* %[[ptr]], align 4
+// CHECK:  %[[ptr:.+]] = getelementptr inbounds %struct.RECORD.0, %struct.RECORD.0 addrspace(6)* %[[node_ptr]], i32 0, i32 3, i32 0
+// CHECK:  store i32 %[[tid_group_x]], i32 addrspace(6)* %[[ptr]], align 4
+// CHECK:  %[[ptr:.+]] = getelementptr inbounds %struct.RECORD.0, %struct.RECORD.0 addrspace(6)* %[[node_ptr]], i32 0, i32 3, i32 1
+// CHECK:  store i32 %[[tid_group_y]], i32 addrspace(6)* %[[ptr]], align 4
+// CHECK:  %[[ptr:.+]] = getelementptr inbounds %struct.RECORD.0, %struct.RECORD.0 addrspace(6)* %[[node_ptr]], i32 0, i32 3, i32 2
+// CHECK:  store i32 %[[tid_group_z]], i32 addrspace(6)* %[[ptr]], align 4
+// CHECK:  ret void
+
+struct RECORD
+{
+  uint3 dtid;
+  uint3 gid;
+  uint gidx;
+  uint3 gtid;
+};
+
+[Shader("node")]
+[NodeLaunch("coalescing")]
+[numthreads(4,4,4)]
+void node01(RWGroupNodeInputRecords<RECORD> input,
+ uint3 DTID : SV_DispatchThreadID,
+ uint3 GID : SV_GroupID,
+ uint GIdx : SV_GroupIndex,
+ uint3 GTID : SV_GroupThreadID )
+{
+  input.Get().dtid = DTID;
```

Comment:
Since this shader has total numthreads > 1, most writes to the RWGroupNodeInputRecords are data races.  This doesn't matter right now because we don't check for things like this yet, but if we were to (like we should), each of these tests would be expected to fail for this reason.  Perhaps we should change the code to be conditional on `GIdx == 0`, which would mean they get written only on the first thread in the group, and are no longer a data race.

Reply:
conditional on GIdx == 0.